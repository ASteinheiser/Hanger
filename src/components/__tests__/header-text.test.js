import React from 'react';

import HeaderText from '../header-text.js';

describe('HeaderText tests ->', () => {

  describe('Basic tests ->', () => {

    it('renders a header text component', () => {
      const component = shallow(
        <HeaderText />
      );
      expect(component.props().size).toEqual('36px');
      expect(component.props().children).toEqual(undefined);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Advanced tests ->', () => {

    it('does not pass props...', () => {
      const component = shallow(
        <HeaderText someprops='hello!'/>
      );
      expect(component.props().someprops).toEqual(undefined);
      expect(component).toMatchSnapshot();
    });

    it('except for the \'small\' prop', () => {
      const component = shallow(
        <HeaderText small={true} />
      );
      expect(component.props().size).toEqual('28px');
      expect(component).toMatchSnapshot();
    });

    it('and the \'text\' prop', () => {
      const component = shallow(
        <HeaderText text={'hello world!'} />
      );
      expect(component.props().children).toEqual('hello world!');
      expect(component).toMatchSnapshot();
    });
  });
});
