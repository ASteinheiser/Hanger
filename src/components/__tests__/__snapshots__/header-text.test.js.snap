// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderText tests -> Advanced tests -> and the 'text' prop 1`] = `
<Styled(Text)
  color="#4976f9"
  size="36px"
>
  hello world!
</Styled(Text)>
`;

exports[`HeaderText tests -> Advanced tests -> does not pass props... 1`] = `
<Styled(Text)
  color="#4976f9"
  size="36px"
/>
`;

exports[`HeaderText tests -> Advanced tests -> except for the 'small' prop 1`] = `
<Styled(Text)
  color="#4976f9"
  size="28px"
/>
`;

exports[`HeaderText tests -> Basic tests -> renders a header text component 1`] = `
<Styled(Text)
  color="#4976f9"
  size="36px"
/>
`;
